pepr/privacy/mia.py

```diff
@@ -0,0 +1,1182 @@
+
+        Returns
+        -------
+        list
+            List of shape (target_models, classes) containing the classified
+            predictions per target model and data class.
+        """
+        pred_list = []
+        for i, target_model in enumerate(target_models):
+            pred_list.append([])
+            for j in range(len(target_data["indices_per_target"][i])):
+                class_indices = target_data["indices_per_target"][i][j]
+                pred_list[i].append(
+                    target_model.predict(attack_test_data[class_indices])
+                )
+
+        return pred_list
+
+    @staticmethod
+    def _attack_model_evaluation(
+        attack_models,
+        target_predictions,
+        target_data,
+    ):
+        """
+        Evaluate attack models.
+
+        Parameters
+        ----------
+        attack_models : iterable
+            List of trained attack models to evaluate.
+        target_predictions : list
+            List of prediction vectors per target model and attack model.
+        target_data : dict
+            Dictionary storing the classified indices and labels of target training and
+            evaluation records.
+
+        Returns
+        -------
+        dict
+            Dictionary storing the attack model results. A list "per attack model and
+            target model" has the shape (attack model, target model) -> First index
+            specifies the attack model, the second index the target model.
+
+            * test_accuracy_list (numpy.ndarray): Evaluation accuracy on test data per
+              attack model and target model.
+            * precision_list (numpy.ndarray): Attack precision per attack model and
+              target model.
+            * recall_list (numpy.ndarray): Attack recall per attack model and target
+              model.
+            * tp_list (numpy.ndarray): True positives per attack model and target model.
+            * fp_list (numpy.ndarray): False positives per attack model and target
+              model.
+            * fn_list (numpy.ndarray): False negatives per attack model and target
+              model.
+            * tn_list (numpy.ndarray): True negatives per attack model and target model.
+            * test_accuracy (numpy.ndarray): Evaluation accuracy averaged over all
+              attack models per target model.
+            * precision (numpy.ndarray): Attack precision averaged over all attack
+              models per target model.
+            * recall (numpy.ndarray): Attack recall averaged over all attack models per
+              target model.
+            * overall_test_accuracy (float): Evaluation accuracy averaged over all
+              attack models and target models.
+            * overall_precision (float): Attack precision averaged over all attack
+              models and target models.
+            * overall_recall (float): Attack recall averaged over all attack models and
+              target models.
```

Comment:
If we change the variable naming of the `attack_results` dict we also should rename these variables the same way.

Reply:
Done in 03c76ef